.github/workflows/typechecks.yml

```diff
@@ -0,0 +1,52 @@
+name: Tests, Linter & Typecheck
+
+on:  
+  push:
+    branches:
+      - main
+  pull_request: 
+    types: 
+      - opened
```

Comment:
```suggestion
  pull_request:
```

Reply:
To ensure this runs when new commits are added.

Reply:
Done